plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -245,6 +228,40 @@ public MachineSource saveToSnapshot() throws MachineException {
         }
     }
 
+    /**
+     * Creates snapshot of workspace and push it to remote docker registry.
+     * This method has protected access because it can be extended to improve flexibility.
+     *
+     * @param image
+     *         snapshot image name (may include namespace)
+     * @param registry
+     *         registry to which snapshot image should be pushed
+     * @return Docker machine source
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    protected DockerMachineSource createAndPushSnapshotToRemoteRegistry(String image, String registry) throws IOException,
+                                                                                                              InterruptedException {
+        PushParams pushParams = PushParams.create(image)
+                                          .withRegistry(registry)
+                                          .withTag(LATEST_TAG);
+
+        final String fullRepo = pushParams.getFullRepo();
+        commitContainer(fullRepo, LATEST_TAG);
+        //TODO fix this workaround. Docker image is not visible after commit when using swarm
```

Comment:
If you think that it should be done by separate issue, can you please create an issue for that?
Because i don't think that such kind of TODO is really ok + `Thread.sleep(2000)` doesn't look okay either